packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -98,7 +98,9 @@ const getOverlayAlpha = (elevation: number) => {
 };
 
 const getBoxShadowColor = (theme: Theme) => {
-  return alpha(theme.palette.common.black, 0.21);
+  return theme.vars
+    ? `rgba(${theme.vars.palette.common.black} /  0.21)`
```

Comment:
```suggestion
    ? `rgba(0 0 0 /  0.21)`
```
This is not perfect but I think it works fine for 99.99% because `common.black` does not change between color modes.